### PR TITLE
feature: introduces better markdown functionality for posts #221

### PR DESCRIPTION
**Pull request by elliotBraem**
Makes changes to post form to feature a more powerful markdown editor

_Resolves https://github.com/near/neardevhub-widgets/issues/221_

### CODE REVIEW

**Comment by elliotBraem** [comment]
Acceptance Criteria:

- [x] Text editor for posts in feed, view [here](https://near.org/devhub.efiz.near/widget/gigs-board.pages.Feed?nearDevGovGigsContractAccountId=devgovgigs.near&nearDevGovGigsWidgetsAccountId=devhub.efiz.near).
- [x] Creator for ideas/solutions, view [here](https://near.org/devhub.efiz.near/widget/gigs-board.pages.Create?nearDevGovGigsContractAccountId=devgovgigs.near&nearDevGovGigsWidgetsAccountId=devhub.efiz.near).
- [x] Wiki editor for communities, view [here](https://near.org/devhub.efiz.near/widget/gigs-board.pages.community.new?nearDevGovGigsContractAccountId=devgovgigs.near&nearDevGovGigsWidgetsAccountId=devhub.efiz.near).
- [x] Live previews are still visible below as they were before.
- [x] Line breaks format on separate lines.


Notes:
- Checkboxes are a custom implementation and there are some known inconsistencies:
    - the <Markdown> component in the VM renders checkboxes with a prepended bullet (may be undesirable?)
    - pressing enter for a new line after creating a checkbox does not create another checkbox (like on github). This should be possible, but requires some more savvy development... will see if I can resolve)
- "Reference to other [thing]" is the icon to the far right. It will write "bos://" to text and relies on this [PR to the VM](https://github.com/NearSocial/VM/pull/90/commits) to be merged before working. But once it works, we can embed widgets and other things.
- The full SimpleMDE widget can be found [here](https://near.social/devhub.efiz.near/widget/SimpleMDE). You can see all of the available functionalities and the line/word counter in the bottom right. Let me know if any of these should be incorporated into DevHub. Icons can be interchanged with the icons found [here](https://fontawesome.com/v4/icons/).
- This was a larger story than I anticipated :)

File: src/gigs-board/components/organism/editor.jsx
Comment (carina-akaia) [inline src/gigs-board/components/organism/editor.jsx:223]: Let's remove this class and pass it only in particular cases when it's really needed

File: src/gigs-board/components/organism/editor.jsx
Comment (carina-akaia) [inline src/gigs-board/components/organism/editor.jsx:280]: See the previous comment

**Review by carina-akaia — CHANGES_REQUESTED** [review]

**Comment by elliotBraem** [comment]
After finishing the development for posts and applying it to community pages, I'd like to move community page editor to a separate PR for #178. Community pages (Wiki + Bio) has some additional pieces to it that require more attention.

You can see the current work [here](https://near.org/devhub.efiz.near/widget/gigs-board.pages.community.new?nearDevGovGigsContractAccountId=devgovgigs.near&nearDevGovGigsWidgetsAccountId=devhub.efiz.near), but notice how the label does not show up. This will require some thinking.

Otherwsie, this PR is ready for review.

File: src/gigs-board/components/molecule/markdown-editor.jsx
Comment (ailisp) [inline src/gigs-board/components/molecule/markdown-editor.jsx:15]: Tried out every format hear. Majority of them looks great. The `quote` style, doesn't look obvious that it means to quote something:
image

Can we tweak the style a little bit?

File: src/gigs-board/components/molecule/markdown-editor.jsx
Comment (ailisp) [inline src/gigs-board/components/molecule/markdown-editor.jsx:21]: mention used to have an auto-completion. Is it possible to keep that?
image

**Review by ailisp — APPROVED** [review]
Looks great! And code is very clean and neat!

**Comment by ailisp** [comment]
Mention looks good, but quote still looks like just bigger sized text:
image

image

Is it possible to make it looks like github quote:
image

**Comment by elliotBraem** [comment]
This is ready for merge now @ailisp @ori-near

**Comment by ori-near** [comment]
Thanks @elliotBraem – what link should I use to see the latest preview?

**Comment by ori-near** [comment]
Hi @elliotBraem, thanks so much for posting this. Overall looks great. Here are few suggestions I found. I would say number 1 is the most important. We can create a separate ticket for any of the other ones if they're not quick fixes. Thank you.

**1: Line Breaks**
Entering separate line breaks do not render that way. Instead, they concatenate.
Screenshot 2023-09-06 at 9 49 57 AM

**2: URLS**
Clicking on the URL icon shows `[](http://)`. Could we change that to `[](https://)`? Also, it seems like when I just enter www.devhub.org the URL doesn't work. 

**3: @ Handles**
When I click or type the @ handle, once I start typing text such as `Ell`, I see the the list of suggested names. When I click Elliot, the description only shows `Ell` but the preview shows the full name. Once a user is selected, can we show the full name also in the description?
Screenshot 2023-09-06 at 9 56 07 AM

**4: Quotes**
Right now quote CSS displays the text as larger, which makes it hard to distinguish. Is there a way to style quote text similar to how GitHub or Gmail displays them:
Screenshot 2023-09-06 at 9 58 39 AM

**5: Code Block**
Could we style code blocks inside a gray box or something, to help make it visually different from other text?
Screenshot 2023-09-06 at 10 01 03 AM

**6: Reference Thing**
What's the purpose of the reference thing? I wasn't sure actually how to use it. Do we need it?

**Comment by elliotBraem** [comment]
🤦 

Check out the below, how's this?

Screenshot 2023-09-06 at 9 20 41 PM

The only one I had trouble with was mentions... which was definitely a crazy ride to figure out -- but we did it! I can deploy a preview again, but I currently have #245 deployed -- I can do again after that's merged if you'd like.

As for "reference thing", I have removed the button for now, but when we move to the social contract, we can enable some really cool functionalities like how we reference pull requests and commits on Github... and so many other things 😄

**Comment by ailisp** [comment]
Code looks pretty good! From the screenshot I can see a lot of points from Ori is addressed and the remaining points at least you seem to tested them (for 1,2 and 3). Given this PR has been blocked by other PRs for quite long I'm going to merge it.

**Comment by ailisp** [comment]
Double checked in production, everything points from Ori is addressed! Great work!

**Pull request merged**
